lms/static/js/components/card/views/card.js

```diff
@@ -0,0 +1,89 @@
+ * - description (string or function): Description of the card. Defaults to the empty string.
+ * - details (array or function): Array of child views to be rendered as details of this card. The class "meta-detail"
+ *      is automatically added to each rendered child view to ensure appropriate styling. Defaults to an empty list.
+ * - actionClass (string or function): Class name for the action DOM element. Defaults to the empty string.
+ * - actionUrl (string or function): URL to navigate to when the action button is clicked. Defaults to the empty string.
+ * - actionContent: Content of the action button. This may include HTML. Default to the empty string.
+ */
+;(function (define) {
+    'use strict';
+    define(['jquery',
+            'backbone',
+            'text!templates/components/card/square_card.underscore',
+            'text!templates/components/card/list_card.underscore'],
+        function ($, Backbone, squareCardTemplate, listCardTemplate) {
+            var CardView = Backbone.View.extend({
+                events: {
+                    'click .action' : 'action'
+                },
+
+                switchOnConfiguration: function (square_result, list_result) {
+                    return (this.callIfFunction(this.configuration) || 'square_card') === 'square_card' ?
+                        square_result : list_result;
+                },
+
+                callIfFunction: function (value) {
+                    if ($.isFunction(value)) {
+                        return value.call(this);
+                    } else {
+                        return value;
+                    }
+                },
+
+                initialize: function () {
+                    this.template = this.switchOnConfiguration(
+                        _.template(squareCardTemplate),
+                        _.template(listCardTemplate)
+                    );
+                    this.render();
+                },
+
+                className: function () {
+                    return 'card ' +
+                        this.switchOnConfiguration('square-card', 'list-card') +
+                        ' ' + this.callIfFunction(this.cardClass);
+                },
+
+                configuration: function () {
+                    return this.options.configuration;
+                },
+
+                render: function () {
+                    this.$el.html(this.template({
+                        title: this.callIfFunction(this.title),
+                        description: this.callIfFunction(this.description),
+                        action_class: this.callIfFunction(this.actionClass),
+                        action_url: this.callIfFunction(this.actionUrl),
+                        action_content: this.callIfFunction(this.actionContent)
+                    }));
+                    var detailsEl = this.$el.find('.card-meta-details');
+                    _.each(this.callIfFunction(this.details), function (detail) {
```

Comment:
Check out this blog post for a nifty way to render subviews: http://ianstormtaylor.com/rendering-views-in-backbonejs-isnt-always-simple/